### ISSUE TITLE
z2m plugin debug bug

### ISSUE THREAD

**Issue by copystring** [Luligu/matterbridge-zigbee2mqtt#35]
For the debug:
**- the frontend log** has his own system and display only matterbridge and plugins: 
- in the Log page you can not only select the level but also add a search string. The selection there is persisted between session and applies only to Log page. 
![image](https://github.com/Luligu/matterbridge-zigbee2mqtt/assets/132135057/c95e6444-9f8d-4d55-9190-6603d6ee96ad)
- the small log in the Home page shows always everything.

**- the console log** displays the matterbridge and matter.js log with the level you select here or from the command line:
![image](https://github.com/Luligu/matterbridge-zigbee2mqtt/assets/132135057/2cd26dab-275e-4b1e-80b0-b8ad7be1cf17)
**- the console log** also displays the plugin log with or without debug depending on this:
![image](https://github.com/Luligu/matterbridge-zigbee2mqtt/assets/132135057/72dd472d-28d0-4f91-a4ea-deff7db21c6f)

_Originally posted by @Luligu in https://github.com/Luligu/matterbridge-zigbee2mqtt/issues/33#issuecomment-2187392721_

Please take a look at this video and let me know if the debug messages in the home tab are intended.

https://user-images.githubusercontent.com/1298861/342930557-befa3572-582b-4bf7-be4f-c17ca3e9ca91.mp4

**Comment by copystring** [comment]
> * the small log in the Home page shows always everything.

This behavior is very confusing. As you know, I have a lot of devices and the little log shows way too many messages, making it hard to read with all those debug messages I don't need.

From a user experience, there are way too many ways to enable/disable debug logs.

If you ask me, one log type should be globally selected and then one for each plugin. Everything else is too much.

**Comment by copystring** [comment]
Try this video because the upload to GitHub did not work.
https://youtu.be/vJrLQ3LRnrI

**Comment by Luligu** [comment]
What would you like to have and not to have?
What would you like to change?

**Comment by copystring** [comment]
One option to globally change the log type and one option to enable/disable debug for each plugin should be enough.
I think this could best be done in the settings tab of matterbridge or in the log tab, but not both, or at the least make the two settings synchronized. Having it in both tabs is what I think is confusing.

Another thing that confused me is the log in the home tab shows different content than the log in the log tab. How would any user know this is the expected behavior? The log in the home tab does not have much room and sometimes the debug messages use up a lot of room making it hard to follow. See my example below.
Maybe the log area taller if a user really wants debug messages in the home tab.

![image](https://github.com/Luligu/matterbridge-zigbee2mqtt/assets/1298861/7d69ba8d-d70c-43cf-955f-121e39479d48)

**Comment by Luligu** [comment]
Hi,
I finally finished the refactor of loggers.
It will be in the latest soon (needs a few publish from all plugins so it will take a few days..).
If something else is not clear in a week open another issue please.
Thanks.

**Comment by copystring** [comment]
Thank you! I'll have a look when you're finished
